test/lit/merge/renamings.wat

```diff
@@ -0,0 +1,360 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-merge %s first %s.second second -all -S -o - | filecheck %s
+
+;; Test that we rename items in the second module to avoid name collisions.
+
+(module
+  ;; CHECK:      (type $array (array (mut funcref)))
+  (type $array (array (mut (ref null func))))
+
+  ;; This tag has a conflict in second.wat, and so second.wat's $foo
+  ;; will be renamed.
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (type $ref|$array|_=>_none (func (param (ref $array))))
+
+  ;; CHECK:      (type $i32_=>_none (func (param i32)))
+
+  ;; CHECK:      (type $i64_=>_none (func (param i64)))
+
+  ;; CHECK:      (type $f32_=>_none (func (param f32)))
+
+  ;; CHECK:      (type $f64_=>_none (func (param f64)))
+
+  ;; CHECK:      (global $foo i32 (i32.const 1))
+
+  ;; CHECK:      (global $bar i32 (i32.const 2))
+
+  ;; CHECK:      (global $other i32 (i32.const 3))
+
+  ;; CHECK:      (global $bar_2 i32 (i32.const 4))
+
+  ;; CHECK:      (memory $foo 10 20)
+
+  ;; CHECK:      (memory $bar 30 40)
+
+  ;; CHECK:      (memory $foo_2 50 60)
+
+  ;; CHECK:      (memory $other 70 80)
+
+  ;; CHECK:      (data $foo (i32.const 1) "abc")
+
+  ;; CHECK:      (data $bar (i32.const 2) "def")
+
+  ;; CHECK:      (data $other (memory $foo_2) (i32.const 3) "ghi")
+
+  ;; CHECK:      (data $bar_2 (memory $foo_2) (i32.const 4) "jkl")
+
+  ;; CHECK:      (table $foo 10 20 funcref)
+
+  ;; CHECK:      (table $bar 30 40 funcref)
+
+  ;; CHECK:      (table $foo_2 50 60 funcref)
+
+  ;; CHECK:      (table $other 70 80 funcref)
+
+  ;; CHECK:      (elem $foo func $foo $bar)
+
+  ;; CHECK:      (elem $bar func $bar $foo)
+
+  ;; CHECK:      (elem $other func $foo_3 $other)
+
+  ;; CHECK:      (elem $bar_2 func $other $foo_3)
+
+  ;; CHECK:      (tag $foo (param i32))
+  (tag $foo (param i32))
+
+  ;; CHECK:      (tag $bar (param i64))
+  (tag $bar (param i64))
+
+  ;; This memory has a conflict in second.wat, and so second.wat's $foo
+  ;; will be renamed.
+  (memory $foo 10 20)
+
+  (memory $bar 30 40)
```

Comment:
If you reorder the module elements to match the output order, the check lines will match up better.

Reply:
It's annoying though because sometimes after I do it it gets reordered again anyhow by the updater...

Rather than us working on this manually each time it would probably be good to improve the updater script?

Reply:
The updater is just using the tool output and not reordering anything itself, so the question is why `Print.cpp` isn't giving you a stable order.

Reply:
Ah, it could be Print. But I believe Print is deterministic, so I suspect it's something in the interaction between the two tools. See the last two commits: the one before last is a manual reordering, and the last is the effects of the auto-updater, which didn't change it as much as I worried, but still causes enough reordering that I'm not sure it's better overall.